tests/routes_tests/visitor_name.test.js

```diff
@@ -0,0 +1,66 @@
+const test = require('tape');
+const request = require('supertest');
+const app = require('../../react-backend/app');
+const jwt = require('jsonwebtoken');
+
+test('POST /api/user/name-from-scan | viable & registered hash', (t) => {
+  const token = jwt.sign({ email: 'jinglis12@googlemail.com' }, process.env.SECRET);
+
+  const successPayload = {
+    hashToCheck:
+      '9fb59d630d2fb12f7478c56c5f1b2fff20e0dd7c9d3a260eee7308a8eb6cd955',
+  };
+  request(app)
+    .post('/api/user/name-from-scan')
+    .set('authorization', token)
+    .send(successPayload)
+    .expect(200)
+    .expect('Content-Type', /json/)
+    .expect(res => res.json().fullname)
+    .end((err, res) => {
+      t.ok(res === 'james bond', 'Hash is viable and found in database');
```

Comment:
Instead of using `t.ok` to check if two things are equal, you should use the provided `t.equal`. It's more readable, and gives you more useful output in case of failure compared to the `expected: true, actual: false` you're getting now.

Reply:
When debugging, your job is to trace the problem back from where it appears to where is has originated from. Your instinct should always be to think about how to try and get more information on what is being returned or calculated at every step. 

In your case, your assertion is expecting `true`, but receiving a `false`. This is because the comparison fails. Your first port of call should be to look at what you're actually getting back in `res` and see what that tells you about where in the backend code the response must have come from.

Reply:
ah thanks, I'll do that. Though now I think I'm getting too much information haha

Reply:
Just to join in, I feel like line 19 should be completed in the `.end` to be more readable  . It feels weird using `.expect` like middleware.

Reply:
Yeah turns out it wasn't actually doing anything anyway so I've taken it out